fineract-provider/src/main/java/org/apache/fineract/infrastructure/jobs/domain/JobExecutionRepository.java

```diff
@@ -165,30 +154,82 @@ public void updateJobStatusToFailed(Long stuckJobId, String partitionerStepName)
-            return "SELECT cjp.id FROM batch_custom_job_parameters cjp WHERE JSON_CONTAINS(cjp.parameter_json,:jsonString)";
-        } else if (databaseTypeResolver.isPostgreSQL()) {
-            return "SELECT cjp.id FROM (SELECT id,json_array_elements(parameter_json) AS json_data FROM batch_custom_job_parameters) AS cjp WHERE (cjp.json_data ::jsonb @> :jsonString ::jsonb)";
-        } else {
-            throw new IllegalStateException("Database type is not supported for json query " + databaseTypeResolver.databaseType());
+    public LocalDate getBusinessDateOfRunningJobByExecutionParameter(String jobName, String jobCustomParamKeyName, String parameterKeyName,
+            String parameterValue, String dateParameterName) {
+        try {
+            if (databaseTypeResolver.isPostgreSQL()) {
+                return namedParameterJdbcTemplate.queryForObject("""
+                        SELECT JSON ->> 'parameterValue' FROM
+                        json_array_elements((SELECT
+                            CJP.parameter_json
+                        FROM
+                            BATCH_JOB_INSTANCE BJI
+                                INNER JOIN BATCH_JOB_EXECUTION BJE ON BJI.JOB_INSTANCE_ID = BJE.JOB_INSTANCE_ID
+                                INNER JOIN BATCH_JOB_EXECUTION_PARAMS BJEP ON BJE.JOB_EXECUTION_ID = BJEP.JOB_EXECUTION_ID
+                                inner join batch_custom_job_parameters CJP ON cast(BJEP.parameter_value as bigint) = CJP.id
+                                AND BJEP.parameter_name = :jobCustomParamKeyName
+                                CROSS JOIN LATERAL json_array_elements(CJP.parameter_json) J
+                        WHERE
+                                    J ->> 'parameterName' = :parameterKeyName
+                          AND J ->> 'parameterValue' = :parameterValue
+                          AND BJE.STATUS IN (:statuses)
+                          AND BJI.JOB_NAME = :jobName
+                          AND BJE.JOB_INSTANCE_ID NOT IN (
+                            SELECT
+                                IBJE.JOB_INSTANCE_ID
+                            FROM
+                                BATCH_JOB_INSTANCE IBJI
+                                    INNER JOIN BATCH_JOB_EXECUTION IBJE ON IBJI.JOB_INSTANCE_ID = IBJE.JOB_INSTANCE_ID
+                            WHERE
+                                    IBJE.STATUS IN (:completedStatuses)
+                              AND IBJI.JOB_NAME = :jobName
+                        ))) JSON WHERE JSON ->> 'parameterName' = :dateParameterName;
+                        """,
+                        Map.of("jobCustomParamKeyName", jobCustomParamKeyName, "parameterKeyName", parameterKeyName, "parameterValue",
+                                parameterValue, "statuses", List.of(STARTED.name(), STARTING.name()), "completedStatuses",
+                                List.of(COMPLETED.name(), FAILED.name(), UNKNOWN.name()), "jobName", jobName, "dateParameterName",
```

Comment:
good catch, should be a copy-paste error, I'm fixing it

Reply:
i am not saying its wrong...is it?

Reply:
it is, because if have a 'STARTED' and a 'FAILED' status for the same instance, then most probably the first execution failed  and the current execution has the 'started' status.

Reply:
is it possible to have 2 STARTED job run but one of them has FAILED instance status and the other one has COMPLETED and in that case it will return it is running but it is not, no?

Reply:
if 1 of them is COMPLETED, then we will never take it as a running one, because the inner query.